lib/deprecated/zbuff_compress.c

```diff
@@ -72,14 +72,36 @@ size_t ZBUFF_compressInit_advanced(ZBUFF_CCtx* zbc,
                                    const void* dict, size_t dictSize,
                                    ZSTD_parameters params, unsigned long long pledgedSrcSize)
 {
+    size_t ret;
     if (pledgedSrcSize==0) pledgedSrcSize = ZSTD_CONTENTSIZE_UNKNOWN;  /* preserve "0 == unknown" behavior */
-    return ZSTD_initCStream_advanced(zbc, dict, dictSize, params, pledgedSrcSize);
+    ret = ZSTD_CCtx_reset(zbc, ZSTD_reset_session_only);
+    ret = ZSTD_isError(ret) ? ret : ZSTD_CCtx_setPledgedSrcSize(zbc, pledgedSrcSize);
```

Comment:
minor : a macro would be helpful here to reduce repetition

Reply:
Sure. I can use `FORWARD_IF_ERROR()` if we're comfortable taking an internal dependency on zstd. Otherwise I can gin up my own.

Reply:
It's reasonable to depend on `lib/common` for this code